common/src/main/java/com/redhat/lightblue/rest/ExternalResourceConfiguration.java

```diff
@@ -0,0 +1,112 @@
+package com.redhat.lightblue.rest;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.redhat.lightblue.util.JsonInitializable;
+
+/**
+ * <p>Parses a configuration file containing paths to external jar files to be loaded into memory
+ * at runtime.</p>
+ *
+ * <p>
+ * Expected format is:
+ * [
+ *    "path/to/file.jar",
+ *    "path/to/directory/"
+ * ]
+ * </p>
+ *
+ * @author dcrissman
+ */
+public class ExternalResourceConfiguration implements JsonInitializable {
+
+    public final static String PROTOCOL_FILE = "file";
+
+    private final Set<URL> externalPaths = new HashSet<>();
+
+    public Set<URL> getExternalUrls() {
+        return Collections.unmodifiableSet(externalPaths);
+    }
+
+    public ExternalResourceConfiguration() {}
+
+    public ExternalResourceConfiguration(JsonNode node) throws IOException {
```

Comment:
What's the point of initializing this from a jsonnode?  Maybe some documentation around this if we expect to read in a json file or something.